backend/dataall/modules/datapipelines/cdk/datapipelines_pipeline.py

```diff
@@ -71,6 +72,27 @@ def get_env_team(self, pipeline: DataPipeline) -> EnvironmentGroup:
             env = EnvironmentService.get_environment_group(session, pipeline.SamlGroupName, pipeline.environmentUri)
         return env
 
+    def create_pipeline_artifacts_bucket(self, artifact_bucket_base_name: str):
+        artifact_bucket_key = kms.Key(
+            self,
+            f'{artifact_bucket_base_name}-key',
+            removal_policy=RemovalPolicy.RETAIN,
+            alias=f'{artifact_bucket_base_name}-key',
+            enable_key_rotation=True,
+        )
+        artifact_bucket = s3.Bucket(
+            self,
+            f'{artifact_bucket_base_name}-bucket',
+            bucket_name=f'{artifact_bucket_base_name}-bucket',
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+            removal_policy=RemovalPolicy.RETAIN,
```

Comment:
nit: Do we really want `removal_policy=RemovalPolicy.RETAIN,`? When we delete a pipeline stack I think the artefacts bucket is no longer used. wdyt?